ledger/catchpointtracker.go

```diff
@@ -474,14 +476,8 @@ func doRepackCatchpoint(header CatchpointFileHeader, in *tar.Reader, out *tar.Wr
 			return err
 		}
 
-		buf := make([]byte, header.Size)
-		bytesRead := int64(0)
-		for (err == nil) && (bytesRead < header.Size) {
-			var x int
-			x, err = in.Read(buf[bytesRead:])
-			bytesRead += int64(x)
-		}
-		if (err != nil) && (err != io.EOF) {
+		buf, err := io.ReadAll(in)
```

Comment:
I was thinking about this loop and how much GC pressure it makes, how many files/chunks are in a typical catchpoint file? We could move creating `buf` to be outside of the loop and re-used for each chunk file. We could remember what the max capacity would be needed for this bufer needs (from stage 1) and then reset the slice for each iteration `iterBuf = buf[:desiredLen]` to reset length but keep the capacity?

Reply:
It's an interesting idea. Do you think this might be the cause for high memory usage in Pavel's tests?

Reply:
It is a lot of stuff to allocate and destroy all at once.. POC re-use commit https://github.com/cce/go-algorand/commit/9fdebcfd8df94f0aac1885ef00ca6d5dc613def2